app/controllers/admin/documents_controller.rb

```diff
@@ -9,4 +9,19 @@ def by_content_id
       redirect_to url_maker.admin_editions_path
     end
   end
+
+  def edit_needs
+    @edition = Edition.find(params[:edition_id])
+    @document = @edition.document
+  end
+
+  def update_needs
+    @edition = Edition.find(params[:edition_id])
+    document = @edition.document
+
+    document.need_ids = params[:need_ids] || []
+    document.patch_meets_user_needs_links
+
+    redirect_to admin_edition_path(@edition)
+  end
```

Comment:
The REST purist in me wonders if these could be show and update on a needs controller that we hit via `GET /:content_id/needs/` and `PUT /:content_id/needs` ?  It feels slightly odd that we use the edition id in a documents controller.